### PR TITLE
feat: Make kube-apiserver endpoint configurable

### PR DESCRIPTION
**Pull request by sanderterschure**
# Pull Request

## What? (description)
In the current helm chart there is no option to configure the Kubernetes API endpoint when deploying the CCM as a daemonset. This PR adds the option to make the kube-apiserver endpoint configurable. 

```
...
daemonSet:
  enabled: true
  k8s:
    serviceHost: 10.0.0.1
    servicePort: 443
...
```

## Why? (reasoning)
In my setup we are running the kube-apiserver on a different port because of firewall limitations. Since this is not configurable the deployed will fail with the following logs: 
```
0930 13:21:31.971223       1 serving.go:386] Generated self-signed cert in-memory
I0930 13:21:32.407675       1 serving.go:386] Generated self-signed cert in-memory
W0930 13:21:32.407727       1 client_config.go:659] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
unable to load configmap based request-header-client-ca-file: Get "https://10.60.0.11:6443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication": dial tcp 10.60.0.11:6443: connect: connection refused
E0930 13:21:32.776521       1 run.go:72] "command failed" err="unable to load configmap based request-header-client-ca-file: Get \"[https://10.60.0.11:6443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication\](https://10.60.0.11:6443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication/)": dial tcp 10.60.0.11:6443: connect: connection refused"
```

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [x] you linted your code (`make lint`)
- [x] you linted your code (`make unit`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: charts/talos-cloud-controller-manager/values-example.yaml [hunk 1]
```diff
@@ -53,7 +52,8 @@ transformations:
       node-role.kubernetes.io/db: ""
 
 # Deploy the Talos Cloud Controller Manager as a DaemonSet
-useDaemonSet: true
+daemonSet:
+  enabled: true
```
Comment (sergelogvinov) [inline charts/talos-cloud-controller-manager/values-example.yaml:56]: Thank you for your PR!!! 

Can we keep the `daemonSet.enable` option set to `false` by default? It's a rare case when you need the daemonSet setup, and it's only necessary with the `nodeIPAM` controller.

Comment (sanderterschure) [reply charts/talos-cloud-controller-manager/values-example.yaml:56]: Sure, I've kept this default for my setup. Will set default to `false`.

**Comment by sergelogvinov** [comment]
Hi, sorry for delay.

Can you:
* use low case on commit message
* add commit body
* DSO (sign-off) commit (git commit -s) 
* and make one commit (squash commit) in one

And we can merge it! 👍

**Review by sergelogvinov — APPROVED** [review]

**Pull request merged**